ts/session/utils/job_runners/jobs/GroupSyncJob.ts

```diff
@@ -0,0 +1,239 @@
+    const namespace = change.pushed.namespace;
+    switch (namespace) {
+      case SnodeNamespaces.ClosedGroupInfo: {
+        if (change.pushed.seqno) {
+          toConfirm[1].groupInfo = [change.pushed.seqno.toNumber(), change.updatedHash];
+        }
+        break;
+      }
+      case SnodeNamespaces.ClosedGroupMembers: {
+        toConfirm[1].groupMember = [change.pushed.seqno.toNumber(), change.updatedHash];
+        break;
+      }
+      case SnodeNamespaces.ClosedGroupKeys: {
+        // TODO chunk 2 closed group
+        break;
+      }
+      default:
+        assertUnreachable(namespace, 'buildAndSaveDumpsToDB assertUnreachable');
+    }
+  }
+
+  await MetaGroupWrapperActions.metaConfirmPushed(...toConfirm);
+  return LibSessionUtil.saveDumpsToDb(groupPk);
+}
+
+async function pushChangesToGroupSwarmIfNeeded(groupPk: GroupPubkeyType): Promise<RunJobResult> {
+  // save the dumps to DB even before trying to push them, so at least we have an up to date dumps in the DB in case of crash, no network etc
+  await LibSessionUtil.saveDumpsToDb(groupPk);
+  const changesToPush = await LibSessionUtil.pendingChangesForGroup(groupPk);
+  // If there are no pending changes then the job can just complete (next time something
+  // is updated we want to try and run immediately so don't schedule another run in this case)
+  if (isEmpty(changesToPush?.messages)) {
+    return RunJobResult.Success;
+  }
+
+  const msgs: Array<StoreOnNodeData> = changesToPush.messages.map(item => {
+    return {
+      namespace: item.namespace,
+      pubkey: groupPk,
+      networkTimestamp: GetNetworkTime.getNowWithNetworkOffset(),
+      ttl: TTL_DEFAULT.TTL_CONFIG,
+      data: item.ciphertext,
+    };
+  });
+
+  const result = await MessageSender.sendEncryptedDataToSnode(
+    msgs,
+    groupPk,
+    changesToPush.allOldHashes
+  );
+
+  const expectedReplyLength =
+    changesToPush.messages.length + (changesToPush.allOldHashes.size ? 1 : 0);
+
+  // we do a sequence call here. If we do not have the right expected number of results, consider it a failure
+  if (!isArray(result) || result.length !== expectedReplyLength) {
+    window.log.info(
+      `GroupSyncJob: unexpected result length: expected ${expectedReplyLength} but got ${result?.length}`
+    );
+
+    // this might be a 421 error (already handled) so let's retry this request a little bit later
+    return RunJobResult.RetryJobIfPossible;
+  }
+
+  const changes = LibSessionUtil.batchResultsToGroupSuccessfulChange(result, changesToPush);
+  if (isEmpty(changes)) {
+    return RunJobResult.RetryJobIfPossible;
+  }
+  // Now that we have the successful changes, we need to mark them as pushed and
+  // generate any config dumps which need to be stored
+
+  await confirmPushedAndDump(changes, groupPk);
+  return RunJobResult.Success;
+}
+
+class GroupSyncJob extends PersistedJob<GroupSyncPersistedData> {
+  constructor({
+    identifier, // this has to be the pubkey to which we
```

Comment:
which we what?

Reply:
fixed to  `   identifier, // this has to be the groupPubkey`